core/src/service/share_service.rs

```diff
@@ -120,6 +127,10 @@ impl RequestContext<'_, '_> {
         if file.owner() == owner {
             return Err(CoreError::FileNotShared);
         }
+        // must have access to file
+        if tree.access_mode(owner, &id)?.is_none() {
+            return Err(CoreError::FileNonexistent); // todo: put this shit everywhere
```

Comment:
@Parth something to think about - this check probably belongs everywhere the deleted checks are